lib/busbar_cli/services/logs.rb

```diff
@@ -1,14 +1,72 @@
 module Services
   class Logs
-    def self.call(container_id:, environment_name:, since:)
+    def self.call(app_id:, environment_name:, component_id:, since:, size:)
       Services::Kube.setup
+      require 'open3'
 
-      container_name = container_id.split('-')[0..-3].join('-')
+      @environment_name = environment_name
+      @app_id = app_id
+      @component_id = component_id
 
-      Kernel.exec(
-        "#{KUBECTL} --context=#{Services::Kube.current_profile} " \
-        "logs -f --since=#{since} #{container_id} #{container_name} -n #{environment_name}"
+      # Retrieve Component Data
+      # TODO: Refactor this into a service => https://github.com/busbar-io/busbar-cli/issues/45
+      if @component_id.nil? then
+        # retrieve app components from server
+        environment_data = JSON.parse(EnvironmentsRepository.get(environment_name, app_id))
+        components_data = environment_data['data']['components']
+
+        if components_data.length == 1 then
+          @component_id = components_data[0]['type']
+        else
+          puts "More than one component found for #{@app_id} #{@environment_name}."
+          puts "Available components:"
+          components_data.each do |component|
+            puts "  #{component['type']}"
+          end
+          puts "Please chose one from the list above and try again."
+          return
+        end
+      end
+
+      # Retrieve a Pod Name
+      stdout, stderr, status = Open3.capture3(
+        "#{KUBECTL} --context=#{Services::Kube.current_profile} --namespace #{environment_name} " \
+        "get pod -l busbar.io/app=#{app_id},busbar.io/component=#{@component_id} | grep -v '^NAME' | head -n 1 | awk '{print $1}'"
+      )
+
+      if status.to_s.end_with?('0') and stdout.length > 0 then
+        pod_name = stdout.to_s.chomp
+      else
+        puts "Application not found. Exiting!"
```

Comment:
Is `Exiting!` necessary? I would feel bad enough if I typed something wrong and got "Application not found." There's no need to rub it in. 😅 

```suggestion
        puts "Application not found."
```